src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1738,6 +1796,48 @@ skip:
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  void t_cryptonote_protocol_handler<t_core>::select_peer_for_testing(uint64_t block_height, const cryptonote::blobdata &block, const boost::uuids::uuid &except, boost::uuids::uuid &selected)
+  {
+    std::vector<boost::uuids::uuid> peers;
+    m_p2p->for_each_connection([&](cryptonote_connection_context& ctx, nodetool::peerid_type peer_id, uint32_t support_flags)->bool{
+      if (ctx.m_connection_id == except)
+        return true;
+      if (ctx.m_remote_blockchain_height < block_height)
+        return true;
+      if (ctx.m_state < cryptonote_connection_context::state_normal)
+        return true;
+      if (ctx.m_waiting_for_block != "")
+        return true;
+      peers.push_back(ctx.m_connection_id);
+      return true;
+    });
+    if (peers.empty())
+      return;
```

Comment:
This can stop block propagation in the edge case where we have 2 peers connected to each other and no one else. It's safer to select peer for testing when there's more than 1 to select from.

Reply:
Good point, I'll make it three, since with 2 it's 50% chance to select one, and the score change will average at 0, risking -2 easily.